### PR TITLE
Block guideline updates

### PR DESCRIPTION
**Pull request by tellyworth**
Some proposed updates, still WIP.

### CODE REVIEW

File: blocks.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+<h4>3. Plugin and block names should reflect the block's purpose.</h4>
+
+Plugin titles and block titles should describe what the block does in a way that helps users easily understand its purpose. In most cases the plugin title and the block title should be identical or very similar. Some examples of good plugin and block names include:
+
+`Rainbow Block`
+`Sepia Image Grid`
+`Business Hours Block`
+
+Please avoid plugin and block titles unrelated to the block itself, or that cannot be easily distinguished from core blocks. Some examples of unhelpful plugin and block names are things such as:
```
Comment (kraftbj) [inline blocks.md:60]: What about a case like "Jetpack Business Hours" for a "Business Hours" block made by Jetpack?

File: blocks.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+They should not rely on an external API or cloud service for functions that could be performed locally.
+
+<h4>7. Server-side code should be kept to a minimum.</h4>
+
+Since Block Plugins are WordPress plugins, they necessarily contain PHP code for metadata and initialisation. As much as possible, they should avoid including additional PHP code or server-side libraries. The WordPress REST API should be the preferred interface to WordPress, rather than custom server-side code.
+
+We recognise that there are limits to the REST API, and situations where server-side PHP is the only performant way to implement a feature. In those situations, PHP code may be included. Please ensure that such code is clearly written, used as little as possible, and well documented.
+
+<h4>8. Must not include advertisements or promote other blocks, plugins, or themes.</h4>
```
Comment (kraftbj) [inline blocks.md]: There was a conversation yesterday in #meta that I wanted to get clarity on in line with the guidlines.

https://wordpress.slack.com/archives/C02QB8GMM/p1594850305004600

Using my "Jetpack Business Hours" example. That block is currently part of Jetpack, but it is relatively trivial for it to live in a single/ad-hoc block plugin.

With the assumption that we handle a case where someone has both installed, would it be allowed under these guidelines for it to live in both Jetpack and in an ad-hoc plugin.

For the sake of the conversation, I'm not concerned with it potentially being extra work for the developer to keep in sync or whether it is practical :), just would that ad hoc plugin be allowed while the identical block is still within a block collection/other plugin.

Comment (Ipstenu) [reply blocks.md]: Currently allowed, but **strongly** recommended against, with the expectation that eventually you would remove it from Jetpack. And it would be disallowed for new plugins going the other way.

We don't want some developer to have 30 plugins, where 29 are ad-hoc-blocks and 1 is a collection of all their blocks. That's just gaming the system, making more work for themselves (you don't care, fine, but other people really need to remember that they're not Jetpack/large company with many hands making light work), and making it a right bloody mess for your users who are trying to find a plugin.

Basically ... if you're going to decouple your blocks, fully decouple them. We'd give allowances for people who (like you) had blocks included in a larger plugin but wanted to break them out later (since you pre-dated the advent of this sub-directory) but the end goal needs to be an either/or, not a both.

File: blocks.md (hunk 1 above)
Comment (Ipstenu) [reply blocks.md:60]: I would say "Business Hours by Jetpack" would be more descriptive :) Same SEO Juice I believe.

File: blocks.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* Blocks will appear in their block library after installation and activation.
+
+<h3>Developer Expectations</h3>
+
+* Developers will have a concrete set of requirements and guidelines to follow when writing blocks for the Block Directory.
+* Following these guidelines will help ensure that the blocks they develop can be seamlessly installed in the editor.
+
+<h3>Definitions</h3>
+
+First up, some clarity. For the purposes of the Block Directory, we distinguish between two types of plugins:
```
Comment (Ipstenu) [inline blocks.md]: I would remove the first sentence.

File: blocks.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+1. Plugins that exist solely to distribute a block.
+1. All other plugins, including those that bundle many independent blocks, plugins that contain blocks in addition to other functionality, and plugins with no blocks at all.
+
+These guidelines apply specifically to the first type of plugin, which we’ll call here a Block Plugin. If your plugin is of the second type, then the further guidelines and restrictions do not apply to your work. All plugins, be they block-only or not, are required to comply with the Detailed Plugin Guidelines https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/.
+
+<h3>Block Plugins and the Block Directory</h3>
+
+The Block Directory contains only Block Plugins, that is to say plugins that contain only a single independent block and a minimum of supporting code. Block plugins have the following characteristics:
```
Comment (Ipstenu) [inline blocks.md]: "The Block Directory contains only Block Plugins which contain a single, independent, top-level block and a minimum of supporting code. Block plugins have the following characteristics:"

I say 'top-level' here to help assuage fears about nested blocks (like Columns would be a single top-level that has 'children' blocks)

File: blocks.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+1. A specific type of WordPress plugin, with the same structure including a readme.txt file.
+1. Containing only blocks (typically one).
```
Comment (Ipstenu) [inline blocks.md]: Containing only blocks (typically a single top-level block)

File: blocks.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+1. Contain a minimum of server-side code.
```
Comment (Ipstenu) [inline blocks.md]: "Containing a minimum of server-side (i.e. PHP) code."

Grammar note, trying to be consistent :)

File: blocks.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+These guidelines apply specifically to the first type of plugin, which we’ll call here a Block Plugin. If your plugin is of the second type, then the further guidelines and restrictions do not apply to your work. All plugins, be they block-only or not, are required to comply with the Detailed Plugin Guidelines https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/.
+
+<h3>Block Plugins and the Block Directory</h3>
+
+The Block Directory contains only Block Plugins, that is to say plugins that contain only a single independent block and a minimum of supporting code. Block plugins have the following characteristics:
+
+1. A specific type of WordPress plugin, with the same structure including a readme.txt file.
+1. Containing only blocks (typically one).
+1. Contain a minimum of server-side code. 
+1. Do not have any UI outside of the post editor.
```
Comment (Ipstenu) [inline blocks.md]: "Self-contained with no UI outside of the post editor."

File: blocks.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+In addition to following the guidelines that apply to all WordPress plugins, Block Plugins that are submitted to the Block Directory must adhere to these guidelines:
```
Comment (Ipstenu) [inline blocks.md]: "In addition to the guidelines that apply to all WordPress plugins, Block Plugins that are submitted to the Block Directory must adhere to these guidelines:"

There's something about having 'following' in there that made it read weird.

File: blocks.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+<h3>Guidelines</h3>
+
+<h4>1. Block Plugins are for the Block Editor.</h4>
+
+Your block plugins should contain a block, and a minimum of other supporting code. It should not contain any UX outside of the editor, such as WordPress options or `wp-admin` menus. Server-side code should be kept to a minimum.
```
Comment (Ipstenu) [inline blocks.md]: "Block plugins must contain a block, and a minimum of other supporting code. It may not contain any UX outside of the editor, such as WordPress options or `wp-admin` menus. Server-side code should be kept to a minimum."

Stronger words here. Since we want blocks to 'just work' (no APIs no registration etc) we need to be firm here. No ~capes~ settings pages.

File: blocks.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+While plugins that provide styles for or extend existing core blocks are permitted, please be aware that they are not yet supported by the Block Editor's seamless install, which means they cannot be included in the Block Directory just yet.
```
Comment (Ipstenu) [inline blocks.md]: "Plugins that extend or provide styles for other, pre-existing, plugins are currently _not_ eligible for inclusion in the Block Directory. At this time, they are not supported by the Block Editor's seamless install process."

File: blocks.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+<h4>2. Block Plugins are separate blocks.</h4>
+
+Block plugins are intended to be single purpose, separate, independent blocks, not bundles or compendiums of many blocks. In most cases, a Block Plugin should contain only one single block. It should not contain multiple blocks that could reasonably be split up into separate, independent plugins.
```
Comment (Ipstenu) [inline blocks.md]: "Block plugins are intended to be single purpose, separate, independent blocks, not bundles or compendiums of many blocks. In most cases, a Block Plugin should contain only one single top-level block. The Block Directory will not include collections of blocks that could be reasonably split up into separate, independent, blocks."

File: blocks.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+`Sepia Image Grid`
+`Business Hours Block`
+
+Please avoid plugin and block titles unrelated to the block itself, or that cannot be easily distinguished from core blocks. Some examples of unhelpful plugin and block names are things such as:
+
+`PluginCo Inc Block`
+`Widget`
+`Image Block`
+
+You must refrain from names and titles that include trademarks or names that you do not legally represent; for example you cannot name a block "Facerange Block" unless you work for Facerange and have their permission.
```
Comment (Ipstenu) [inline blocks.md]: "The same trademark restrictions for plugin titles exist for block titles. This means that a block may not be named "Facerange Block" unless developed by a legal representative of Facerange. For example, if a block is meant to show custom discount information for an ecommerce plugin, then a good name would be "Custom Discounts for WeeCommerce."

We don't use 'you' in other places, so let's try to be consistent :) Tone helps translators!

File: blocks.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+The block name (meaning the [`name` parameter to `registerBlockType()`](https://developer.wordpress.org/block-editor/developers/block-api/block-registration/#block-name) and [`name` in `block.json`](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#name)) must be unique to your block. As with titles, please avoid names containing trademarks or names you do not legally represent.
+
+The namespace prefix to the block name should reflect either the plugin author, or the plugin slug. For example:
+
+`name: "my-rainbow-block-plugin/rainbow"`, or
+`name: "john-doe/rainbow"`, or
+`name: "pluginco/rainbow"`.
+
+The namespace must not be a reserved one such as `core`.
```
Comment (Ipstenu) [inline blocks.md]: "The namespace may not be a reserved one, such as `core` or `wordpress`."

May not. Be firm! Don't break core! :)

File: blocks.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+This includes payment for premium features. You may use the donation link feature.
+
+Block plugins that do require a paid service or include upselling and premium features are still permitted in the main WordPress Plugin Directory, subject to its guidelines.
+
+<h4>6. Block Plugins should work seamlessly.</h4>
+
+Block Plugins are intended to work seamlessly and instantly when installed from the editor. That means they should not encumber the user with additional steps or prerequisites such as installing another plugin or theme, signing up for an account, or logging in or manually connecting to an external service.
+
+Block Plugins must not require another plugin in order to function.
```
Comment (Ipstenu) [inline blocks.md]: Will we revisit this later? This means people can't include WooCommerce blocks, as that requires another plugin. (We really need a plugin dependency check like we have for themes)

File: blocks.md [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+
+They may use an external/cloud API where necessary, provided it can be done seamlessly without requiring a login or activation key.
+
+They should not rely on an external API or cloud service for functions that could be performed locally.
+
+<h4>7. Server-side code should be kept to a minimum.</h4>
+
+Since Block Plugins are WordPress plugins, they necessarily contain PHP code for metadata and initialisation. As much as possible, they should avoid including additional PHP code or server-side libraries. The WordPress REST API should be the preferred interface to WordPress, rather than custom server-side code.
+
+We recognise that there are limits to the REST API, and situations where server-side PHP is the only performant way to implement a feature. In those situations, PHP code may be included. Please ensure that such code is clearly written, used as little as possible, and well documented.
```
Comment (Ipstenu) [inline blocks.md]: "There are limits to the REST API, and situations where server-side PHP is the only performant way to implement a feature. In those situations, PHP code may be included, provided it is clearly written, used as little as possible, and well documented."

Removing 'We' voice.

File: blocks.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+<h4>8. Must not include advertisements or promote other blocks, plugins, or themes.</h4>
+
+Block Plugins must not include code that displays alerts, dashboard notifications, or similar messages promoting other plugins, themes, goods, or services. They may include an unobtrusive link identifying the developer or plugin page.
```
Comment (Ipstenu) [inline blocks.md]: "They may include an unobtrusive link identifying the developer or plugin page." 

Okay I have to ask... How? Because you can't have thee link on the front facing side, so are we saying using the block UX (sidebar) is okay for mild self promotion or...?

File: blocks.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+* Developers will have a concrete set of requirements and guidelines to follow when writing blocks for the Block Directory.
+* Following these guidelines will help ensure that the blocks they develop can be seamlessly installed in the editor.
+
+<h3>Definitions</h3>
+
+First up, some clarity. For the purposes of the Block Directory, we distinguish between two types of plugins:
+1. Plugins that exist solely to distribute a block.
+1. All other plugins, including those that bundle many independent blocks, plugins that contain blocks in addition to other functionality, and plugins with no blocks at all.
+
+These guidelines apply specifically to the first type of plugin, which we’ll call here a Block Plugin. If your plugin is of the second type, then the further guidelines and restrictions do not apply to your work. All plugins, be they block-only or not, are required to comply with the Detailed Plugin Guidelines https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/.
```
Comment (Ipstenu) [inline blocks.md]: "These guidelines apply specifically to the first type of plugin, which are called a Block Plugin. If a plugin is of the second type, then the further guidelines and restrictions do not apply. All plugins, be they block-only or not, are required to comply with the Detailed Plugin Guidelines https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/."

File: blocks.md [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+
+`PluginCo Inc Block`
+`Widget`
+`Image Block`
+
+You must refrain from names and titles that include trademarks or names that you do not legally represent; for example you cannot name a block "Facerange Block" unless you work for Facerange and have their permission.
+
+<h4>3a. Block names should be unique and properly namespaced.</h4>
+
+The block name (meaning the [`name` parameter to `registerBlockType()`](https://developer.wordpress.org/block-editor/developers/block-api/block-registration/#block-name) and [`name` in `block.json`](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#name)) must be unique to your block. As with titles, please avoid names containing trademarks or names you do not legally represent.
```
Comment (Ipstenu) [inline blocks.md]: "The block name (meaning the [`name` parameter to `registerBlockType()`](https://developer.wordpress.org/block-editor/developers/block-api/block-registration/#block-name) and [`name` in `block.json`](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#name)) must be unique to the block. As with titles, please respect trademarks and other projects' commonly used names, so as not to conflict with them."

File: blocks.md [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+
+Block Plugins must include a valid `block.json` file. In addition to the requirements specified in the RFC, the `block.json` must include the following attributes:
+* name
+* title
+* At least one of: script, editorScript
+* At least one of: style, editorStyle
+
+<h4>5. Block Plugins must not require payment or a paid account to function.</h4>
+
+This includes payment for premium features. You may use the donation link feature.
```
Comment (Ipstenu) [inline blocks.md]: No form of payment is permitted for the use of a Block Plugin. While it is allowed to use the donation link feature in the plugin's readme, or to link from the full plugin listing, the Block Plugin itself must be free to use.

File: blocks.md (hunk 1 above)
Comment (kraftbj) [reply blocks.md:60]: Thanks for both replies!

File: blocks.md (hunk 10 above)
Comment (tellyworth) [reply blocks.md]: I added this and clarified it a bit further.

File: blocks.md (hunk 12 above)
Comment (tellyworth) [reply blocks.md]: I've added the first sentence. The second seems to suggest that a block that depends on WeeCommerce would be permitted - which currently it would not.

File: blocks.md (hunk 14 above)
Comment (tellyworth) [reply blocks.md]: Until we have a way to specify and search based on dependencies, I think it's important to prohibit plugin dependencies. Otherwise people will wind up installing blocks they can't use.

However, I have expanded and changed some of the clauses around that. In particular I've made it clear that it should be fine to make use of an existing plugin if it happens to be installed, as long as the block is also usable without it.

For example, a "One Day Only Discount Banner Block" plugin could use hooks to interact with WeeCommerce, for example to display a coupon code - as long as that feature is optional and the discount banner block is usable on sites that don't have WeeCommerce installed.

File: blocks.md (hunk 16 above)
Comment (tellyworth) [reply blocks.md]: I've totally rewritten this section.

File: blocks.md (hunk 19 above)
Comment (tellyworth) [reply blocks.md]: I've changed this, though it's now been moved a bit and combined with some other text - still good?

File: blocks.md (hunk 14 above)
Comment (Ipstenu) [reply blocks.md]: > Until we have a way to specify and search based on dependencies, 

Yeah, that's fair. As long as we're keeping it open for if/when we get that :sigh: I really want plugin dependencies like we have for theme.

File: blocks.md (hunk 19 above)
Comment (Ipstenu) [reply blocks.md]: I would try to remove the 'you' usage in the doc.

Comment (tellyworth) [reply blocks.md]: I think all the "you" usage is removed now.

**Comment by lukecarbis** [comment]
Is this the right place to discuss these updates?

> Block Plugins must not include code that displays alerts, dashboard notifications, or similar obtrusive messages unrelated to the purpose of the block.

I hate ads as much as the next guy, but ads in traditional WordPress plugins are not only permitted, they're built into the business models that allow for such an active and dynamic plugin ecosystem.

I'm concerned that by disallowing advertisements, the growth of the Block Directory would be adversley impacted.

Further points for consideration:

> similar obtrusive messages

What about unobtrusive "messages"? Is the intent is to allow for a grey area in which the block plugin reviewers can make judgement calls? (Not a criticism – this could be helpful – but it makes it hard for plugin authors to plan a Freemium model when they're at the mercy of the reviewer).

> unrelated to the purpose of the block

This is vague. Does it allow for adding "greyed out" (inaccessible) premium features to free blocks? Should it?

**Comment by Ipstenu** [comment]
> I'm concerned that by disallowing advertisements, the growth of the Block Directory would be adversley impacted.

Serious ask here, where would a block ONLY plugin put an ad?

Remember we're talking about block-only plugins here, not regular plugins with a bunch of features. The only place I can think of is in the block-sidebar, which would be disallowed with a regular plugin so why would it be allowed here?

**Comment by lukecarbis** [comment]
Personally, I can't think of a subtle place to put an ad – but I've learned not to underestimate the creativeness of advertisers.

I can imagine a "notification" style ad… or one that appears inside the block preview.

Okay… so this is a bit of a tangent – but I've seen blocks in the past (can't remember which) that show an alert in the preview area informing the user that the block requires additional configuration. Imagine a PayPal button block that needs you to add your PayPal email for it to work properly. The notification could say "this block isn't yet configured". Another example is the "placeholder" in an image block prior to adding an image. This sort of design precedent could easily give way to messaging like "Huge savings at Yoast! Black Friday Sale".

> The only place I can think of is in the block-sidebar, which would be disallowed with a regular plugin

Serious question:
Imagine a block that had 3 "styles" available in Free, and an additional 3 in Pro. Could the free version of that block show a dropdown list containing all 6 styles, but 3 were greyed out, with a "helpful" notice saying something like "more styles available with Block Pro"?

Is that an ad? I've definitely seen that done.

Just trying to see where the limits are. I'm coming from a position of a user who hates ads in the admin, as well as a Product guy who understands the need for their careful, unobtrusive use. I've [written about my ideas](https://carb.is/2019/12/advertising-in-the-wordpress-admin/) for what should and shouldn't be allowed.

**Comment by Ipstenu** [comment]
> Okay… so this is a bit of a tangent – but I've seen blocks in the past (can't remember which) that show an alert in the preview area informing the user that the block requires additional configuration. Imagine a PayPal button block that needs you to add your PayPal email for it to work properly. The notification could say "this block isn't yet configured". 

Those would not be permitted by default in the Block _only_ directory. And I'm stressing this for a reason :) If your plugin is not just install-and-go, if it requires additional configurations, then it's not applicable at this time for the block directory. You can be in the regular directory, not the block-only sub-directory.

That restriction is covered in the 6th guideline:

> Block Plugins are intended to work seamlessly and instantly when installed from the editor. That means they should not encumber the user with additional steps or prerequisites such as installing another plugin or theme, signing up for an account, or logging in or manually connecting to an external service.

I think we're okay there :)

> Another example is the "placeholder" in an image block prior to adding an image. This sort of design precedent could easily give way to messaging like "Huge savings at Yoast! Black Friday Sale".

That's actually disallowed overall by the Directory Guidelines today. So no. That's not allowed, block directory or not. Messing with a user's interactions like that is not okay.

> Imagine a block that had 3 "styles" available in Free, and an additional 3 in Pro. Could the free version of that block show a dropdown list containing all 6 styles, but 3 were greyed out, with a "helpful" notice saying something like "more styles available with Block Pro"?

That would be an ad (in this case an Upsell, which is an advertisement of other features). But more over, it goes against the bottom line goal here, which is: The Block Directory is for block _only_ plugins that work the second they're installed. They just _work_. And if you're making them not-work for whatever reason (registration, premium features etc) then you're not a good candidate for the Block Directory. That doesn't mean you can't host your code on .org, just that you can't host it in the _Block_ Directory.

**Comment by lukecarbis** [comment]
Great elaborations and explanations @Ipstenu. Thank you!

I'm hearing that right now, for those thinking about creating a Freemium offering that is block-based, it's best to include that in the Plugin directory, _not_ the Block directory. The reason is that the Block Directory should contain simple, straight-forward, no-fuss blocks that _just work_ as soon as they're added.

Perhaps including further explanation about not seeing advertisements or upsells in the "User Expectations" section of this document might be useful? Additionally, in the "Developer Expectations", I'd suggest making it clear that these guidelines are intentionally stricter than the Plugin Directory guidelines, and it's still possible to submit Block plugins there.

**Comment by tellyworth** [comment]
> I'm hearing that right now, for those thinking about creating a Freemium offering that is block-based, it's best to include that in the Plugin directory, _not_ the Block directory. The reason is that the Block Directory should contain simple, straight-forward, no-fuss blocks that _just work_ as soon as they're added.

Yes you're exactly right on the intention and reasoning.

I don't speak for the plugin review team, but I would note one point here. There is a path forward for commercial developers to work with the Block Directory, which is spelled out in section 5: a Block Plugin can interact with other WordPress plugins, provided the block is still useful without it and the other rules are followed. So a block in the Block Directory could indeed load additional styles or trigger extra functionality from a commercial plugin or theme if it's available on the site. It'll have to abide by the rules about not requiring or upselling that plugin or theme though. I'll defer to @Ipstenu on exactly what constitutes an upsell ad, but I would think that a discreet note in the description that the block is compatible with / works with Plugin X seems like it might be reasonable(?)

> Perhaps including further explanation about not seeing advertisements or upsells in the "User Expectations" section of this document might be useful? Additionally, in the "Developer Expectations", I'd suggest making it clear that these guidelines are intentionally stricter than the Plugin Directory guidelines, and it's still possible to submit Block plugins there.

Thanks for the suggestions! I've added some more clarifying language to the Expectations sections, does that help?

**Comment by lukecarbis** [comment]
@tellyworth Yes, that language works really well, especially the "Developer Expectations" section. Thank you.

File: blocks.md [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+
+`name: "my-rainbow-block-plugin/rainbow"`, or
+`name: "john-doe/rainbow"`, or
+`name: "pluginco/rainbow"`.
+
+The namespace may not be a reserved one such as `core` or `wordpress`.
+
+<h4>4. Block Plugins must include a <code>block.json</code> file.</h4>
+
+The Block Registration RFC outlines the format of a `block.json` file: https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md
```
Comment (coreymckrill) [inline blocks.md]: It appears that this link doesn't exist anymore.

Comment (coreymckrill) [reply blocks.md]: One alternative might be the handbook page for the `registerBlockType` API:
https://developer.wordpress.org/block-editor/developers/block-api/block-registration/

But there isn't anything on that page that draws a direct connection to the block.json file.

Comment (ryelle) [reply blocks.md]: Looks like it was actually moved to this file: https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-metadata.md (see https://github.com/WordPress/gutenberg/pull/23832)

Comment (ryelle) [inline blocks.md]:
```suggestion
The Block Registration RFC outlines the format of a `block.json` file: https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-metadata.md
```

**Comment by jeffpaul** [comment]
With WordPress 5.5 getting released today and the Block Directory officially being a part of core, is there an ETA on getting these guidelines finalized, published, and distributed for developers to utilize as a reference?  Happy to help with any needed copy reviews or testing against existing blocks for questions/concerns/etc.

**Comment by tellyworth** [comment]
>  is there an ETA on getting these guidelines finalized, published, and distributed for developers to utilize as a reference?

The guidelines are published here: https://developer.wordpress.org/plugins/wordpress-org/block-specific-plugin-guidelines/
